components/ws-manager/pkg/manager/monitor.go

```diff
@@ -716,9 +709,7 @@ func (m *Monitor) waitForWorkspaceReady(ctx context.Context, pod *corev1.Pod) (e
 	if err != nil {
 		return xerrors.Errorf("cannot wait for workspace to initialize: %w", err)
 	}
-	m.initializerMapLock.Lock()
-	delete(m.initializerMap, pod.Name)
-	m.initializerMapLock.Unlock()
+	m.clearInitializerFromMap(pod.Name)
```

Comment:
We could consider remove the function `clearInitializerFromMap`, and use the `m.initializerMap.Delete(pod.Name)` instead.

Reply:
We can't do that. This is because there is not always an initializerMap field.

Reply:
@jenting Good point!!! Thanks. I created a PR to address your review.
https://github.com/gitpod-io/gitpod/pull/12286